Wrappers/Python/cil/processors/FluxNormaliser.py

```diff
@@ -0,0 +1,458 @@
+        else:
+            im = ax1.imshow(data_slice.array, cmap='gray',aspect='equal', origin='lower', extent=extent)
+            plt.gcf().colorbar(im, ax=ax1)
+
+        h = data_slice.dimension_labels[1]
+        v = data_slice.dimension_labels[0]
+
+        if h == 'angle':
+            h_min = min_angle
+            h_max = max_angle
+        else:
+            h_min = self.roi[h][0]
+            h_max = self.roi[h][1]
+
+        if v == 'angle':
+            v_min = min_angle
+            v_max = max_angle
+        else:
+            v_min = self.roi[v][0]
+            v_max = self.roi[v][1]
+
+        ax1.plot([h_min, h_max],[v_min, v_min],'--r')
+        ax1.plot([h_min, h_max],[v_max, v_max],'--r')
+
+        ax1.plot([h_min, h_min],[v_min, v_max],'--r')
+        ax1.plot([h_max, h_max],[v_min, v_max],'--r')
+        
+        title = 'ROI'
+        if angle_index is not None:
+            title += ' angle = ' + str(data.geometry.angles[angle_index])
+        if channel_index is not None:
+            title += ' channel = ' + str(channel_index)
+        ax1.set_title(title)
+
+        ax1.set_xlabel(h)
+        ax1.set_ylabel(v)
+
+    @property
+    def num_threads(self):
+        '''Get the optional number of threads parameter to use for the accelerated version.
+
+        Defaults to the value set in the CIL multiprocessing module.'''
+        return cil_mp.NUM_THREADS if self._num_threads is None else self._num_threads
+
+    def set_num_threads(self, value):
+        '''Set the optional number of threads parameter to use for the accelerated version.
+
+        This is discarded if ``accelerated=False``.'''
+        self._num_threads = value
+
+
+        
+    def process(self, out=None):
+        self._calculate_flux()
+        self._calculate_target()
+
+        data = self.get_input()
+        if out is None:
+            out = data.copy()
+
+        proj_size = self.v_size*self.h_size
+        num_proj = int(data.array.size / proj_size)
+        if self._accelerated:
+            num_threads = numba.get_num_threads()
+            numba.set_num_threads(self.num_threads)
+            numba_loop(data.array.ravel(), self.flux.ravel(), self.target_value, num_proj, proj_size, out.array.ravel())
+            # reset the number of threads to the original value
+            numba.set_num_threads(num_threads)
+        else:
+            serial_loop(data.array.ravel(), self.flux.ravel(), self.target_value, num_proj, proj_size, out.array.ravel())
+
+        return out
+
+@numba.njit(parallel=True)
+def numba_loop(arr_flat, flux, target, num_proj, proj_size, out_flat):
+    if len(flux) == 1:
+        norm = target/flux[0]
+        for i in numba.prange(num_proj):
+            proj = arr_flat[i*proj_size:(i+1)*proj_size]*norm
+            out_flat[i*proj_size:(i+1)*proj_size] =  proj
+    else:
+        for i in numba.prange(num_proj):
+            proj = arr_flat[i*proj_size:(i+1)*proj_size]*target/flux[i]
```

Comment:
This is multiplying the array by target, then dividing by flux. It needs some brackets to make sure the array is multiplied directly by a single value - this could be the cause of the timings you see for small datasets.

Reply:
Fixed now